src/runtime_lib/intrinsics.h

```diff
@@ -178,6 +179,47 @@ static int builtin_getVertices(WGraph &edges){
     return edges.num_nodes();
 }
 
+static NodeID builtin_getOutDegree(Graph &edges, NodeID src){
+    return edges.out_degree(src);
+}
+
+static NodeID builtin_getOutDegree(WGraph &edges, NodeID src){
+    return edges.out_degree(src);
+}
+
+static VertexSubset<NodeID>* builtin_getNgh(Graph &edges, NodeID src){
+    auto v =  new VertexSubset<NodeID>(edges.out_degree(src));
+    v->dense_vertex_set_ = (unsigned int*) edges.out_neigh(src).begin();
+    return v;
+}
+
+static VertexSubset<NodeID>* builtin_getNgh(WGraph &edges, NodeID src){
+    auto v =  new VertexSubset<NodeID>(edges.out_degree(src));
+    v->dense_vertex_set_ = (unsigned int*) edges.out_neigh(src).begin();
+    return v;
+}
+
+//static NodeID* builtin_getNgh(Graph &edges, NodeID src){
+//    return edges.out_neigh(src).begin();
+//
+//}
+//
+//static NodeWeight<int, int>* builtin_getNgh(WGraph &edges, NodeID src){
+//    return edges.out_neigh(src).begin();
+//}
+
+static size_t hiroshiVertexIntersection(VertexSubset<NodeID>* A, VertexSubset<NodeID>* B, size_t totalA, size_t totalB, NodeID dest) {
```

Comment:
In the comment, add a reference to the original paper.

Reply:
Hmm this is just a wrapper for hiroshi intersection. Do you think we still need a comment here?